osu.Desktop/DiscordRichPresence.cs

```diff
@@ -114,6 +135,28 @@ private void updateStatus()
                 {
                     presence.Buttons = null;
                 }
+
+                if (!hideIdentifiableInformation && multiplayerClient.Room != null)
+                {
+                    MultiplayerRoom room = multiplayerClient.Room;
+                    presence.Party = new Party
+                    {
+                        Privacy = string.IsNullOrEmpty(room.Settings.Password) ? Party.PrivacySetting.Public : Party.PrivacySetting.Private,
+                        ID = room.RoomID.ToString(),
+                        // technically lobbies can have infinite users, but Discord needs this to be set to something.
+                        // 1024 just happens to look nice.
+                        // https://discord.com/channels/188630481301012481/188630652340404224/1212967974793642034
+                        Max = 1024,
+                        Size = room.Users.Count,
+                    };
+
+                    presence.Secrets.JoinSecret = $"{room.RoomID}:{room.Settings.Password}";
```

Comment:
This is slightly concerning, as it means anyone can hook up a custom Discord RPC client and read the password of any room invitation via the `onJoin` event. While that doesn't mean much, it signaled a flag for me. I'm fine with keeping it as-is since rooms are transient, but needed a confirmation.

Reply:
> anyone can hook up a custom Discord RPC client and read the password of any room invitation via the `onJoin` event

Huh? How does that work?

Reply:
If I understand this correctly, I can run a local version of lazer that logs the `JoinSecret` inside `onJoin`, and then whenever anyone sends me a room invitation, Discord will feed the join message to the currently running modified lazer and I'll get the password from there (unless discord somehow knows I'm not running the correct client).

Reply:
How does stable do this - are room secrets encrypted there as well? I don't recall abuse of this ever really happening, neither do I think it'll be an issue once you do actually have a room password in your hands via sniffing with a custom RPC client or not.

Remember that secrets are only shared when an invite is sent and accepted, so this is on the user in my opinion. If you send an invite (or your room's password) to untrustworthy people, you should expect to get trolled.